src/nina_vesseltun_ex/ex03_project/test_sorting.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+
+__author__ = 'Nina Mariann Vesseltun'
+__email__ = 'nive@nmbu.no'
+
+
+def bubble_sort(datavariable):
+    copy = list(datavariable)
+    for i in range(len(copy) - 1):
+        for j in range(len(copy) - i - 1):
+            if copy[j] > copy[j + 1]:
+                copy[j], copy[j + 1] = copy[j + 1], copy[j]
+    return copy
+
+
+def test_empty():
+    """Test that the sorting function works for empty list"""
+    assert bubble_sort([]) == []
+
+
+def test_single():
+    """Test that the sorting function works for single-element list"""
+    assert bubble_sort([1]) == [1]
+
+
+def test_sorted_is_not_original():
+    """
+    Test that the sorting function returns a new object.
+
+    Consider
+
+    data = [3, 2, 1]
+    sorted_data = bubble_sort(data)
+
+    Now sorted_data shall be a different object than data,
+    not just another name for the same object.
+    """
+    liste = [3, 1, 2]
+    assert id(liste) != id(bubble_sort(liste))
+
+
+def test_original_unchanged():
+    """
+    Test that sorting leaves the original data unchanged.
+
+    Consider
+
+    data = [3, 2, 1]
+    sorted_data = bubble_sort(data)
+
+    Now data shall still contain [3, 2, 1].
+    """
+    data = [3, 1, 2]
+    original = data
+    sorted_data = bubble_sort(data)
+    assert sorted_data is not data
+    assert original is data
+
+
+def test_sort_sorted():
+    """Test that sorting works on sorted data."""
+    data = [3, 1, 2]
+    sorted_data = bubble_sort(data)
+    doubly_sorted_data = bubble_sort(sorted_data)
+    assert doubly_sorted_data == sorted_data
```

Comment:
This just checks that bubble_sort return the same value if the input are lists with same elements and in different orders.

Replace line 64 with `doubly_sorted_data = sorted(data)`